### ISSUE TITLE
Setting up Colliders in Actor onInitialize() does not work

### ISSUE THREAD

**Issue by eonarheim** [excaliburjs/Excalibur#1542]
### Steps to Reproduce
Setup a new collider in an actor onInitialize and collide with another actor
```typescript
class Player extends Actor {
   constructor() {
      super(0, 0, 10, 10);
   }
   onInitialize() {
     this.body.useCircleCollider(10);
   }
}

player.on('precollision', () => {
  console.log("Collides")
})
```

### Expected Result
Precollision event should fire

### Actual Result
Precollision event does not fire

### Environment
- operating system: N/A
- Excalibur versions: master

### Current Workaround
Moving the collider setup to the constructor seems to work around the issue

See: ttps://groups.google.com/d/msgid/excaliburjs/bbb1bd43-c45e-410b-9469-f3eaee7b27b2%40googlegroups.com.

**Comment by catrielmuller** [comment]
I Will try to fix this @eonarheim , to get more knowledge about the core logic.

**Comment by eonarheim** [comment]
@catrielmuller all yours, let me know if you have any questions!

**Comment by catrielmuller** [comment]
@eonarheim Well, I tried to reproduce this issue, but works as expected.

This is an example of my code:
https://gist.github.com/catrielmuller/72b3c8d2fea14bdc8cfba01d5c733d25
![image](https://user-images.githubusercontent.com/2272323/95032219-27ef8d00-0690-11eb-9da3-a729d194c332.png)
![image](https://user-images.githubusercontent.com/2272323/95032245-39d13000-0690-11eb-8f0b-5c19b22b0d32.png)

The thing is that the precollision event it's fired after the broadphase pass.
So basically, if you pair of Objects it's not enough close to intersect the bounds in any way,  this function (https://github.com/excaliburjs/Excalibur/blob/main/src/engine/Collision/DynamicTree.ts#L410) will filter the pair to increase the performance and never will be try to resolve that pair using the collision strategy (That it's where the precollision event it's fired).

**Comment by catrielmuller** [comment]
👍
